thorlcr/activities/loop/thloop.cpp

```diff
@@ -320,9 +321,17 @@ class CGraphLoopActivityMaster : public CLoopActivityMasterBase
         unsigned loopCounter = 1;
         for (;;)
         {
+            Owned<IThorGraphResults> results = queryGraph().createThorGraphResults(1);
+            unsigned condLoopCounter = (helper->getFlags() & IHThorGraphLoopArg::GLFcounter) ? loopCounter : 0;
+            IThorBoundLoopGraph *boundGraph = queryContainer().queryLoopGraph();
+            if (condLoopCounter)
+                boundGraph->prepareCounterResult(*this, results, condLoopCounter, 0);
+
             if (sync(loopCounter))
                 break;
-            queryContainer().queryLoopGraph()->execute(*this, (helper->getFlags() & IHThorGraphLoopArg::GLFcounter)?loopCounter:0, loopResults.get(), extractBuilder.size(), extractBuilder.getbytes());
+
+            boundGraph->queryGraph()->executeChild(extractBuilder.size(), extractBuilder.getbytes(), results, loopResults);
```

Comment:
Previously this would have called graph->setLoopCounter(counter), now it is setting a counter result.  Was it previously wrong?

Reply:
Should be same as before.
It set the loop counter result via prepareCounterResult() and still does (line 243 above).
It used to call setLoopCounter inside the now removed IThorBoundLoopGraph::execute() method, now it is called as part of prepareCounterResult()